src/app/components/fees-row/fees-row.tsx

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { StackProps } from '@stacks/ui';
+import BigNumber from 'bignumber.js';
+import { useField } from 'formik';
+
+import { CryptoCurrencies } from '@shared/models/currencies.model';
+import { FeeTypes, Fees } from '@shared/models/fees/_fees.model';
+import { createMoney } from '@shared/models/money.model';
+import { isNumber, isString } from '@shared/utils';
+
+import { useConvertCryptoCurrencyToFiatAmount } from '@app/common/hooks/use-convert-to-fiat-amount';
+import { convertAmountToBaseUnit } from '@app/common/money/calculate-money';
+import { LoadingRectangle } from '@app/components/loading-rectangle';
+
+import { FeesRowLayout } from './components/fees-row.layout';
+
+interface FeeRowProps extends StackProps {
+  fees?: Fees;
+  isSponsored: boolean;
+}
+export function FeesRow(props: FeeRowProps): JSX.Element {
+  const { fees, isSponsored, ...rest } = props;
+  const [field, _, helper] = useField('fee');
+  const [, __, feeTypeHelper] = useField('feeType');
+  const [fieldWarning, setFieldWarning] = useState<string | undefined>(undefined);
+  const [selected, setSelected] = useState(FeeTypes.Middle);
+  const [isCustom, setIsCustom] = useState(false);
+  const [isSelectVisible, setIsSelectVisible] = useState(false);
```

Comment:
I wonder if we can cut the amount of state here, some of it seems unnecessary/inferable. 

`isCustom` for example? Isn't that just `selected === FeeTypes.Custom`? Why go through the process of setting this value true/false in lines 59 & 63.

Also, `selected`. Isn't this value saved in the form?

Reply:
I can go back thru and refactor the select itself. It does look like `feeType` is the index. I was thinking it was the calculation type for some reason. I wasn't spending much time on it thinking we would prob swap out soon for Radix or other.